### DISCUSSION TITLE
Tabs

### DISCUSSION THREAD

**Discussion by boxadesign** [ONSdigital/design-system#82] [Feedback]
Use this place to discuss the [tabs component](https://ons-design-system.netlify.com/components/tabs) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by bameyrick** [comment]
The background of the tabs appears to be to dark:

<img width="363" alt="Screenshot 2019-07-26 at 12 26 06" src="https://user-images.githubusercontent.com/8447446/61948757-9de26980-afa0-11e9-94f3-760379718f50.png">

**Reply by jrbarnes9** [reply]
Resolved in #593

**Comment by jrbarnes9** [comment]
Typo in tabs attribute...
![Screenshot 2020-06-19 at 11 37 53](https://user-images.githubusercontent.com/43346934/85124398-c8c15600-b221-11ea-937a-8aca319cfa75.png)

**Reply by jrbarnes9** [reply]
Resolved in #793

**Comment by jeremygillies** [comment]
BUG: A bottom border shows on the active tab when there are no tabs in the header. It appears to be affected by padding in the list item of the tabs in the header. I've not figured out a fix. 

https://user-images.githubusercontent.com/1080383/110931635-66196880-8322-11eb-804d-d3280b076611.mp4

**Reply by jeremygillies** [reply]
https://rops-prototypes.netlify.app/prototypes/tabs/

**Reply by jeremygillies** [reply]
Changing height of `.tab--row` from `2.5rem` to `2.55rem` removes the border

**Reply by jrbarnes9** [reply]
Created Issue #1677

**Comment by jrbarnes9** [comment]
The width and position of tabs should remain fixed, regardless of state, and tab content should not be focusable.

Currently, the width of a tab is greater when it is active, which shifts the horizontal position of its siblings. The content of the tab panel can be focussed, which is not necessary.

![tabs-state-issue](https://user-images.githubusercontent.com/43346934/133107625-029c9f1d-0e50-43ab-912d-2beac931ca33.gif)

**Reply by jrbarnes9** [reply]
Issue #1690 raised

**Comment by jrbarnes9** [comment]
Tabs active tab border not hidden in Safari
![Screenshot 2021-10-07 at 13 17 27](https://user-images.githubusercontent.com/43346934/136382630-5b4411b9-7e59-4be4-9649-38a3634d643f.png)

Should be...
<img width="599" alt="Screenshot 2021-10-07 at 13 19 32" src="https://user-images.githubusercontent.com/43346934/136382716-e5c1b343-abe3-4542-b3f6-0b53ab3eb9e7.png">

**Reply by jrbarnes9** [reply]
Raised #1727

**Comment by jrbarnes9** [comment]
JS should check for width of tabs to determine when TOC is used instead

To avoid wrapping tabs
![Screenshot 2022-08-04 at 11 18 37](https://user-images.githubusercontent.com/43346934/182824254-5a439c00-3d3e-456e-810e-c8516b1e380d.png)

**Reply by jrbarnes9** [reply]
Raised #2372

**Comment by jrbarnes9** [comment]
Tab spacing missing

It appears the Service manual is rendering the example correctly, which has highlighted that it's the whitespace in the macro that has been creating the spacing, rather than margins.

![Screenshot 2023-02-08 at 16 50 40](https://user-images.githubusercontent.com/43346934/217597475-d74e442b-7a4d-40a7-899e-59584d922fcd.png)

**Reply by jrbarnes9** [reply]
Raised #2602

**Comment by jrbarnes9** [comment]
The tab labels should be repeated as headings when ToC is used under medium bp.

This would help orientate the user and provide feedback when selecting one of the ToC links.

For example:
![Screenshot 2023-02-08 at 17 59 37](https://user-images.githubusercontent.com/43346934/217613691-365115d6-e863-4db2-9d9c-f5ec45df8793.png)

**Reply by jrbarnes9** [reply]
Raised #2603
